### DISCUSSION TITLE
Allow aliases for @@unique and composite @@id fields

### DISCUSSION THREAD

**Discussion by wadetandy** [prisma/prisma#3732] [Q&A] [answered]
When defining a unique index or composite primary key, prisma introspects the name from the database. You can provide your own name instead, but upon reintrospecting the name will be overwritten. It would be nice to be able to map the name from the database to the prisma schema and resulting types, as this would allow us to name things to better match the domain and intent, especially when applying prisma to legacy databases.

``` prisma
model Thing {
  id Int
  userId Int
  customerId Int

  // Allow a new "map" key to reference the underlying db name, but can use name field for the javascript name
  @@unique([userId, customerId], name: "userAndCustomer", map: "user_customer_unique")

  // Allow a custom name for a composite id filter
  @@id([id, customerId], name: "idAndCustomer")
}
```

``` typescript
prisma.thing.findOne({
  where: {
    // friendlier to our domain AND our eslint rules for camel casing
    userAndCustomer: {
      userId,
      customerId,
   }
 }
})
```

**Comment by ryands17** [comment]
Hey @wadetandy 👋 
It would be great if you could create a feature request [here](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=feature_request.md&title=) so that we can prioritize this.

**Comment by wadetandy** [comment, marked as answer]
#3748 Created, thanks!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
